src/test/java/com/github/commonsrdf/api/AbstractGraphTest.java

```diff
@@ -0,0 +1,192 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.commonsrdf.api;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public abstract class AbstractGraphTest {
+
+	private RDFTermFactory factory;
+	private Graph graph;
+	private IRI alice;
+	private IRI bob;
+	private IRI name;
+	private IRI knows;
+	private IRI member;
+	private BlankNode org1;
+	private BlankNode org2;
+
+	public abstract RDFTermFactory createFactory();
+
+	@Before
+	public void createGraphAndAdd() {
+		factory = createFactory();
+		graph = factory.createGraph();
+		assertEquals(0, graph.size());
+
+		alice = factory.createIRI("http://example.com/alice");
+		bob = factory.createIRI("http://example.com/bob");
+		name = factory.createIRI("http://xmlns.com/foaf/0.1/name");
+		knows = factory.createIRI("http://xmlns.com/foaf/0.1/knows");
+		member = factory.createIRI("http://xmlns.com/foaf/0.1/member");
+		org1 = factory.createBlankNode("org1");
+		org2 = factory.createBlankNode("org2");
+
+		graph.add(alice, name, factory.createLiteral("Alice"));
+		graph.add(alice, knows, bob);
+		graph.add(alice, member, org1);
+		// and for Bob we'll try as Triples
+		graph.add(factory.createTriple(bob, name,
+				factory.createLiteral("Bob", "en-US")));
+		graph.add(factory.createTriple(bob, member, org1));
+		graph.add(factory.createTriple(bob, member, org2));
+
+		graph.add(org1, name, factory.createLiteral("The Secret Club"));
+		graph.add(org1, name, factory.createLiteral("A company"));
+	}
+
+	@Test
+	public void graphToString() {
+
+		System.out.println(graph);
+		assertTrue(graph
+				.toString()
+				.startsWith(
+						"<http://example.com/alice> <http://xmlns.com/foaf/0.1/name> \"Alice\" ."));
+		assertTrue(graph.toString().endsWith(
+				"_:org1 <http://xmlns.com/foaf/0.1/name> \"A company\" ."));
```

Comment:
This needs to be changed to avoid referencing the blank node by identifier here, as serialisers and third-party implementations are free to remap them as necessary. (See other comment about checking/relying on n-triples blank node identifiers.

You could move this test to DummyGraphTest to test its behaviour is consistent with its goals and implementation, but not include it in this abstract test that others will run.